Wait for `theme-directives.html` E2E test to render before taking a screenshot

aloisklink opened the pull request:
## :bookmark_tabs: Summary

The `theme-directives.html` test currently sometimes takes a screenshot before all of the Mermaid diagrams have completed rendering.

This can result in the following error message when running `pnpm run e2e`:

```
1) Configuration and directives - nodes should be light blue
when rendering several diagrams
  diagrams should not taint later diagrams:
Error: Image was 23.384242424242423% different from saved snapshot with 154336 different pixels.
See diff for details: /mermaid/cypress/snapshots/rendering/conf-and-directives.spec.js/__diff_output__/conf-and-directives.spec-when-rendering-several-diagrams-diagram-1.diff.png
at Context.eval (webpack:///./node_modules/.pnpm/cypress-image-snapshot@4.0.1_cypress@12.10.0_jest@29.5.0/node_modules/cypress-image-snapshot/command.js:51:0)
```

![conf-and-directives spec-when-rendering-several-diagrams-diagram-1 diff](https://github.com/mermaid-js/mermaid/assets/19716675/5f6d9f5f-956b-47d5-a837-f81725ef58ba)

## Notes to reviewer

I'm not sure why we're only detecting this issue now in our E2E tests. Maybe https://github.com/mermaid-js/mermaid/pull/4759 changed some performance parameters, and that made the race-condition occur on my PC. Who knows, it could also just be because one of our dependencies updated, or now my PC is overheating and is running slightly slower :laughing:

Also, because this is just a test change, I've labelled this PR as **Skip changelog**, since there's no point in putting it in the release notes.

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [x] :computer: have added necessary unit/e2e tests.
- [ ] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/docs/community/development.md#3-update-documentation) is used for all new features.
- [x] :bookmark: targeted `develop` branch

sidharthv96 reviewed (changes requested):
https://github.com/mermaid-js/mermaid/pull/4847#pullrequestreview-1629893150

aloisklink commented:
> [#4847 (review)](https://github.com/mermaid-js/mermaid/pull/4847#pullrequestreview-1629893150)

Fixed by 17f5052a6f5aee254e8639db6e022349358b862d to use the `urlSnapshotTest()` helper function! I've also removed some of the unused code in the `theme-directives.html` file, since I was making modifications to it anyway.

The only difference is that the snapshot now has a different ID, it's now located at `cypress/snapshots/conf-and-directives.spec.js/Configuration-and-directives---nodes-should-be-light-blue-when-rendering-several-diagrams-diagrams-should-not-taint-later-diagrams.snap.png` and looks like:

![Configuration-and-directives---nodes-should-be-light-blue-when-rendering-several-diagrams-diagrams-should-not-taint-later-diagrams snap](https://github.com/mermaid-js/mermaid/assets/19716675/2a5431ca-a851-453d-a81a-a3e84ddbc88b)
